cmd/acra-translator/grpc_api/server_service_registry.go

```diff
@@ -0,0 +1,32 @@
+package grpc_api
+
+import (
+	"github.com/cossacklabs/acra/cmd/acra-translator/common"
+	"google.golang.org/grpc"
+	"sync"
+)
+
+// GRPCServerSubscriber declares callbacks for gRPC server subscribers
+type GRPCServerSubscriber interface {
+	OnServerInit(server *grpc.Server, data *common.TranslatorData, service DecryptService)
+}
+
+// global registry of gRPC server subscribers
+var grpcServerSubscribers = make([]GRPCServerSubscriber, 0, 8)
+var lock = sync.Mutex{}
+
+// AddgRPCServerSubscriber register callback for gRPC server
+func AddgRPCServerSubscriber(subscriber GRPCServerSubscriber) {
+	lock.Lock()
+	grpcServerSubscribers = append(grpcServerSubscribers, subscriber)
+	lock.Unlock()
+}
+
+// OngRPCServerInit call all registered callbacks on gRPC server initialization
+func OngRPCServerInit(server *grpc.Server, data *common.TranslatorData, service DecryptService) {
+	lock.Lock()
+	defer lock.Unlock()
```

Comment:
Do we really need to use `defer` here? If I am not mistaken the `OnServerInit` function does not return any error so here no possible `early-exit` scenarios, right? Could we place it after the loop statement?

Reply:
Defer has used because of subscribers may panic. Yes, the program will interrupt and exit if nothing catches this panic. But for a future better to handle it from the start correctly.